source/geometry_model/box.cc

```diff
@@ -324,6 +324,24 @@ namespace aspect
       return position_point;
     }
 
+    template <int dim>
+    void
+    Box<dim>::make_periodicity_constraints(AffineConstraints<double> &constraints) const
+    {
+      using periodic_boundary_set
+        = std::set< std::pair< std::pair< types::boundary_id, types::boundary_id>, unsigned int> >;
+      periodic_boundary_set pbs = geometry_model->get_periodic_boundary_pairs();
+
+      for (periodic_boundary_set::iterator p = pbs.begin(); p != pbs.end(); ++p)
+        {
+          DoFTools::make_periodicity_constraints(dof_handler,
```

Comment:
`dof_handler` is a member variable inside the `Simulator` class. That is why it can not be used here. Instead every plugin that is derived from the `SimulatorAccess` class (like this one) can use the functions of that class to get access to these member variables through the functions in include/aspect/simulator_access.h. So in this case `this->get_dof_handler()` has to be used.

Reply:
The error about `DofTools` requires an additional `include` statement at the top of the file. `#include <deal.II/dofs/dof_tools.h>` should do the trick.